File: js/1.java
else if(method.equals("getbankinfo")){
		String ACCOUNTNO = Util.null2String(request.getParameter("ACCOUNTNO")).trim();
		String CURRENCYNO = Util.null2String(request.getParameter("CURRENCYNO"));
		

		LinkedHashMap<String, String> datas = new LinkedHashMap<String, String>();
		datas.put("ACCOUNTNO", ACCOUNTNO);              
		datas.put("CURRENCYNO", CURRENCYNO);  
		JSONArray tmp=new JSONArray();
		tmp.add(datas);
		JSONObject mainObject=new JSONObject();
		mainObject.put("code", "ACCACK");
		mainObject.put("datasource", "OA");
		mainObject.put("traceID", ACCOUNTNO);
		mainObject.put("datas", tmp);
        writeLog("0313search+ACCOUNTNO" +ACCOUNTNO+ mainObject);
		String url="http://192.168.106.118:7001/FDLKF/webservice/rest/TxServiceGateway/gateway";
		JSONObject retobj = new JSONObject();
		HttpPost httpPost = new HttpPost(url);
		CloseableHttpClient client = HttpClients.createDefault();
		//请求参数转JOSN字符串
		StringEntity entity = new StringEntity(mainObject.toString(), "UTF-8");
		entity.setContentEncoding("UTF-8");
		entity.setContentType("application/json");
		httpPost.setEntity(entity);
		try {
			HttpResponse response = client.execute(httpPost);
			if (response.getStatusLine().getStatusCode() == 200) {
				 retobj = JSON.parseObject(EntityUtils.toString(response.getEntity(), "UTF-8"));
			}
		} catch (Exception e) {
			e.printStackTrace();
			retobj.put("error", "连接错误！");

		}

		writeLog("0313search+ACCOUNTNO" +ACCOUNTNO+ retobj);
		if (retobj.getString("success").equals("true")) {//说明接收格式是对的
			JSONArray tmpArray=retobj.getJSONArray("datas");
			//writeLog(tmpArray);
			for (int i = 0; i < tmpArray.size(); i++) {
				String tmpArray_return = tmpArray.getJSONObject(i).getString("RESULT_EXP");
				if (tmpArray_return.equals("查询成功")) {//返回结果, 成功
					String ACCOUNTNO = tmpArray.getJSONObject(i).getString("ACCOUNTNO");
					String CURRENCYNO = tmpArray.getJSONObject(i).getString("CURRENCYNO");
					String CANCELDATE = tmpArray.getJSONObject(i).getString("CANCELDATE");
					String USAGEID = tmpArray.getJSONObject(i).getString("USAGEID");
					String SWIFT_CODE = tmpArray.getJSONObject(i).getString("SWIFT_CODE");
					String ACT_CLTNO = tmpArray.getJSONObject(i).getString("ACT_CLTNO");
					String CTID = tmpArray.getJSONObject(i).getString("CTID");
					String OPENACCOUNTDATE = tmpArray.getJSONObject(i).getString("OPENACCOUNTDATE");
					String FOREIGNTYPE = tmpArray.getJSONObject(i).getString("FOREIGNTYPE");
					String CLTNO = tmpArray.getJSONObject(i).getString("CLTNO");
					String NATUREID = tmpArray.getJSONObject(i).getString("NATUREID");
					String OUT_FLAG = tmpArray.getJSONObject(i).getString("OUT_FLAG");
					String IS_ONLY = tmpArray.getJSONObject(i).getString("IS_ONLY");
					String CANCELREASON = tmpArray.getJSONObject(i).getString("CANCELREASON");
					String UPDATE_TIME = tmpArray.getJSONObject(i).getString("UPDATE_TIME");
					String MEMORYNAME = tmpArray.getJSONObject(i).getString("MEMORYNAME");
					String CNREMARK = tmpArray.getJSONObject(i).getString("CNREMARK");
					String ACCOUNTNAME = tmpArray.getJSONObject(i).getString("ACCOUNTNAME");
					String TO_POS = tmpArray.getJSONObject(i).getString("TO_POS");
					String APPMODE = tmpArray.getJSONObject(i).getString("APPMODE");
					String POSID = tmpArray.getJSONObject(i).getString("POSID");
					String SIGNER = tmpArray.getJSONObject(i).getString("SIGNER");
					String CNAPS_NAME = tmpArray.getJSONObject(i).getString("CNAPS_NAME");
					String ISGRP = tmpArray.getJSONObject(i).getString("ISGRP");
					String RESULT_EXP = tmpArray.getJSONObject(i).getString("RESULT_EXP");
					String CNAPS_CODE = tmpArray.getJSONObject(i).getString("CNAPS_CODE");


					rel.put("ACCOUNTNO", ACCOUNTNO);
					rel.put("CURRENCYNO", CURRENCYNO);
					rel.put("CANCELDATE", CANCELDATE);
					rel.put("USAGEID", USAGEID);
					rel.put("SWIFT_CODE", SWIFT_CODE);
					rel.put("ACT_CLTNO", ACT_CLTNO);
					rel.put("CTID", CTID);
					rel.put("OPENACCOUNTDATE", OPENACCOUNTDATE);
					rel.put("FOREIGNTYPE", FOREIGNTYPE);
					rel.put("CLTNO", CLTNO);
					rel.put("NATUREID", NATUREID);
					rel.put("OUT_FLAG", OUT_FLAG);
					rel.put("OUT_FLAG", OUT_FLAG);
					rel.put("IS_ONLY", IS_ONLY);
					rel.put("CANCELREASON", CANCELREASON);
					rel.put("UPDATE_TIME", UPDATE_TIME);
					rel.put("CNREMARK", CNREMARK);
					rel.put("ACCOUNTNAME", ACCOUNTNAME);
					rel.put("TO_POS", TO_POS);
					rel.put("APPMODE", APPMODE);
					rel.put("POSID", POSID);
					rel.put("SIGNER", SIGNER);
					rel.put("ISGRP", ISGRP);
					rel.put("RESULT_EXP", RESULT_EXP);
					rel.put("CNAPS_CODE", CNAPS_CODE);

				}else  {//信息异常
					String RESULT_EXP = tmpArray.getJSONObject(i).getString("RESULT_EXP");
					rel.put("RESULT_EXP", RESULT_EXP);
				}
			
			}
		}
		
		out.println(rel);
	} 

File: js/1.go
package main

import (
	"bytes"
	"encoding/json"
	"fmt"
	"io"
	"io/ioutil"
	"log"
	"net/http"
)

type ALLmsg struct {
	Code       string  `json:"code"`
	Datasource string  `json:"datasource"`
	TraceID    string  `json:"traceID"`
	INmsg      []INmsg `json:"datas"`
}
type INmsg struct {
	CLTNO           string `json:"CLTNO"`
	ACTCLTNO        string `json:"ACT_ CLTNO"`
	ISGRP           string `json:"ISGRP"`
	APPMODE         string `json:"APPMODE"`
	SIGNER          string `json:"SIGNER"`
	TOPOS           string `json:"TO_POS"`
	ISONLY          string `json:"IS_ONLY"`
	POSID           string `json:"POSID"`
	OUTFLAG         string `json:"OUT_FLAG"`
	CNAPSCODE       string `json:"CNAPS_CODE"`
	SWIFTCODE       string `json:"SWIFT_CODE"`
	CNAPSNAME       string `json:"CNAPS_NAME"`
	OPENACCOUNTDATE string `json:"OPENACCOUNTDATE"`
	ACCOUNTNO       string `json:"ACCOUNTNO"`
	ACCOUNTNAME     string `json:"ACCOUNTNAME"`
	MEMORYNAME      string `json:"MEMORYNAME"`
	CURRENCYNO      string `json:"CURRENCYNO"`
	USAGEID         string `json:"USAGEID"`
	CTID            string `json:"CTID"`
	NATUREID        string `json:"NATUREID"`
	FOREIGNTYPE     string `json:"FOREIGNTYPE"`
	CNREMARK        string `json:"CNREMARK"`
	CANCELDATE      string `json:"CANCELDATE"`
	CANCELREASON    string `json:"CANCELREASON"`
	UPDATETIME      string `json:"UPDATE_TIME"`
	ACCOUNTTYPE     string `json:"ACCOUNT_TYPE"`
}

type RootName struct {
	Datas      []Datas `json:"datas"`
	Total      int     `json:"total"`
	TraceID    string  `json:"traceID"`
	Code       string  `json:"code"`
	Success    bool    `json:"success"`
	Msg        string  `json:"msg"`
	Datasource string  `json:"datasource"`
}
type Datas struct {
	CURRENCYNO string `json:"CURRENCYNO"`
	ACCOUNTNO  string `json:"ACCOUNTNO"`
	RETMSG     string `json:"RET_MSG"`
	RETURN     int    `json:"RETURN"`
}

func testPost() {

	// requestBody := fmt.Sprintf(`{
	requestBody := `{
		"code": "ACCAPPLY",
		"datasource": "OA",
		"traceID": "5641153615",
		"datas": [
			{
				"CLTNO": "001",
				"ACT_CLTNO": "aa212154",
				"ISGRP": "1",
				"APPMODE": "biu一声授权",
				"SIGNER": "李有权#李无权",
				"TO_POS": 1,
				"IS_ONLY": "1",
				"POSID": "12333",
				"OUT_FLAG": "1",
				"CNAPS_CODE": "302100011018",
				"SWIFT_CODE": "",
				"CNAPS_NAME": "中信银行安哥拉总行11111",
				"OPENACCOUNTDATE": "2022-05-11",
				"ACCOUNTNO": "a10102",
				"ACCOUNTNAME": "jc_00001",
				"MEMORYNAME": "jc_00001",
				"CURRENCYNO": "CNY",
				"USAGEID": "1",
				"CTID": "101",
				"NATUREID": "1",
				"FOREIGNTYPE": "1",
				"CNREMARK": "修改了",
				"CANCELDATE": "",
				"CANCELREASON": "",
				"UPDATE_TIME": "2022-05-18",
				"ACCOUNT_TYPE": "0"
			}
		]
	}`

	var jsonStr = []byte(requestBody)
	// jsonStr := strings.NewReader(requestBody)
	url := "http://192.168.106.118:7001/FDLKF/webservice/rest/TxServiceGateway/gateway"
	req, err := http.NewRequest("POST", url, bytes.NewBuffer(jsonStr))
	req.Header.Set("Content-Type", "application/json")
	client := &http.Client{}
	resp, err := client.Do(req)
	if err != nil {
		panic(err)
	}
	defer resp.Body.Close()
	// fmt.Println("response Status:", resp.Status)
	// fmt.Println("response Headers:", resp.Header)
	body, _ := ioutil.ReadAll(resp.Body)

	fmt.Println("response Body:", string(body))
	rel := RootName{}
	jsonerr := json.Unmarshal(body, &rel)
	if jsonerr != nil {
		log.Fatal(jsonerr)
	}
	fmt.Println(rel.Success)
	fmt.Println(rel.Datas[0].RETMSG)
}

func testPost2() {
	ba := ALLmsg{}
	ba.Code = "ACCAPPLY"
	ba.Datasource = "OA"
	ba.TraceID = "55555555"
	inm := INmsg{}
	inm.CLTNO = "001"
	inm.ACTCLTNO = "77777"
	inm.ISGRP = "1"
	inm.APPMODE = "df"
	inm.SIGNER = "呜呜呜"
	inm.TOPOS = "1"
	inm.ISONLY = "1"
	inm.POSID = "1111111111111111"
	inm.OUTFLAG = "1"
	inm.CNAPSCODE = "302100011018"
	inm.SWIFTCODE = ""
	inm.CNAPSNAME = "todo"
	inm.OPENACCOUNTDATE = "2022-05-20"
	inm.ACCOUNTNO = "999881"
	inm.ACCOUNTNAME = "a999888"
	inm.MEMORYNAME = "a999888"
	inm.CURRENCYNO = "CNY"
	inm.USAGEID = "1"
	inm.CTID = "102"
	inm.NATUREID = "1"
	inm.FOREIGNTYPE = "1"
	inm.CNREMARK = "add"
	inm.CANCELDATE = ""
	inm.CANCELREASON = ""
	inm.UPDATETIME = "2022-05-23"
	inm.ACCOUNTTYPE = "1"
	ba.INmsg = append(ba.INmsg, inm)

	requestBody := new(bytes.Buffer)
	json.NewEncoder(requestBody).Encode(ba)
	url := "http://192.168.106.118:7001/FDLKF/webservice/rest/TxServiceGateway/gateway"
	req, err := http.NewRequest("POST", url, requestBody)
	req.Header.Set("Content-Type", "application/json")
	client := &http.Client{}
	resp, err := client.Do(req)
	if err != nil {
		panic(err)
	}
	defer resp.Body.Close()
	fmt.Println(resp.StatusCode)
	// fmt.Println("response Status:", resp.Status)
	// fmt.Println("response Headers:", resp.Header)
	body, _ := io.ReadAll(resp.Body)
	fmt.Println("response Body:", string(body))
	rel := RootName{}
	jsonerr := json.Unmarshal(body, &rel)
	if jsonerr != nil {
		log.Fatal(jsonerr)
	}
	fmt.Println(rel.Success)
	fmt.Println(rel.Datas[0].RETMSG)
}

func main() {
	testPost2()
}
